sklearn/ensemble/gradient_boosting.py

```diff
@@ -1545,17 +1537,13 @@ def _fit_stages(self, X, y, y_pred, sample_weight, random_state,
                                      X_csc, X_csr)
 
             # track deviance (= loss)
+            self.train_score_[i] = loss_(y[sample_mask], y_pred[sample_mask],
```

Comment:
to avoid making a copy of the data, you might use `sample_mask = slice(None, None)`

Reply:
I reverted the changes, thanks for the warning. Using `slice` would break at some other places.